flocker_bb/ec2.py

```diff
@@ -121,29 +120,15 @@ def __init__(self, access_key, secret_access_token, region):
             inputContext={}, world=MethodSuffixOutputer(self))
 
     def identifier(self):
-        return self.name
+        return self.driver.name
 
     def output_START(self, context):
         """
         Create a node.
         """
         def thread_start():
-            image = get_image(
-                self._driver, self.image_name, self.image_tags)
-            self.image_metadata = {
-                'image_id': image.id,
-                'image_name': image.name,
-            }
-            self.image_metadata.update(image.extra['tags'])
-            return self._driver.create_node(
-                name=self.name,
-                size=get_size(self._driver, self.size),
-                image=image,
-                ex_keyname=self.keyname,
-                ex_userdata=self.userdata,
-                ex_metadata=self.instance_tags,
-                ex_securitygroup=self.security_groups,
-            )
+            self.image_metadata = self.driver.get_image_metadata()
```

Comment:
I think there is potentially a race here, between getting the metadata and getting the image (so that the metadata may potentially refer to a different image than is actually started. I think it would make sense to refactor the code so that the metadata is extracted from the image that is actually used to create the instance.